pkg/cloudprovider/vsphere/route/routeprovider.go

```diff
@@ -0,0 +1,182 @@
+package route
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/route/config"
+	"k8s.io/klog"
+)
+
+type RouteProvider struct {
+	routerPath string
+	broker     *nsxtBroker
+	kubeClient clientset.Interface
+}
+
+// NewRouteProvider creates a new RouteProvider
+func NewRouteProvider(cfg *config.Config) (*RouteProvider, error) {
+	if cfg == nil || cfg.Route.RouterPath == "" {
+		return nil, nil
+	}
+	nsxtbroker, err := NewNsxtBroker(&cfg.NSXT)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating nsxt broker failed")
+	}
+	return &RouteProvider{
+		broker:     nsxtbroker,
+		routerPath: cfg.Route.RouterPath,
+	}, nil
+}
+
+// Initialize sets kubernetes client in RouteProvider
+func (p *RouteProvider) Initialize(client clientset.Interface) {
+	p.kubeClient = client
+}
+
+// ListRoutes returns a list of routes which have static routes on NSXT
+func (p *RouteProvider) ListRoutes(ctx context.Context, clusterName string) ([]*cloudprovider.Route, error) {
```

Comment:
Any chance we can add some unit tests for this method as well as `CreateRoute` and `DeleteRoute`?

Reply:
OK, let me try to mock NSXAPI response